tracer/src/Datadog.Trace/Debugger/Symbols/SymbolsUploader.cs

```diff
@@ -0,0 +1,288 @@
+               (settings.SymbolDatabaseUploadEnabled ||
+                (EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.Debugger.SymbolDatabaseUploadEnabledInternal, "false").ToBoolean() ?? false)))
+            {
+                return new SymbolsUploader(api, discoveryService, settings, tracerSettings, serviceName);
+            }
+
+            Log.Information("Symbol database uploading is disabled. To enable it, please set {EnvironmentVariable} environment variable to 'true'.", ConfigurationKeys.Debugger.SymbolDatabaseUploadEnabled);
+            return new NoOpUploader();
+        }
+
+        private void RegisterToAssemblyLoadEvent()
+        {
+            AppDomain.CurrentDomain.AssemblyLoad += async (_, args) =>
+            {
+                await ProcessItemAsync(args.LoadedAssembly).ConfigureAwait(false);
+            };
+        }
+
+        private async Task ProcessItemAsync(Assembly assembly)
+        {
+            await Task.Yield();
+            await _assemblySemaphore.WaitAsync(_cancellationToken.Token).ConfigureAwait(false);
+
+            if (_cancellationToken.IsCancellationRequested)
+            {
+                return;
+            }
+
+            try
+            {
+                await ProcessItem(assembly).ConfigureAwait(false);
+            }
+            finally
+            {
+                _assemblySemaphore.Release();
+            }
+        }
+
+        private async Task ProcessItem(Assembly assembly)
+        {
+            try
+            {
+                if (AssemblyFilter.ShouldSkipAssembly(assembly))
+                {
+                    return;
+                }
+
+                if (!_alreadyProcessed.Add(assembly.Location))
+                {
+                    return;
+                }
+
+                await UploadAssemblySymbols(assembly).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error while trying to extract assembly symbol {Assembly}", assembly);
+            }
+        }
+
+        private async Task UploadAssemblySymbols(Assembly assembly)
+        {
+            using var symbolsExtractor = SymbolExtractor.Create(assembly);
+            if (symbolsExtractor == null)
+            {
+                return;
+            }
+
+            var assemblyScope = symbolsExtractor.GetAssemblySymbol();
+
+            var root = new Root
+            {
+                Service = _serviceName,
+                Env = _environment,
+                Language = "dotnet",
+                Version = _serviceVersion,
+                Scopes = new[] { assemblyScope }
+            };
+
+            await UploadClasses(root, symbolsExtractor.GetClassSymbols()).ConfigureAwait(false);
+        }
+
+        private async Task UploadClasses(Root root, IEnumerable<Model.Scope?> classes)
+        {
+            var count = 0;
+            var builder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
```

Comment:
Yes you right, although the builder size is in characters count and the size limit is bytes count but still will be better to use this instead of MaxBuilderSize. Fixed in d0cc18f5c1ad4c87320df52fbff2fb56d19bef1a

Reply:
Your comment is very legit, I would initialize it then like by using `_thresholdInBytes / sizeof(char)`.
Feel free to resolve.